public/components/custom_panels/helpers/utils.tsx

```diff
@@ -384,6 +391,183 @@ export const renderCatalogVisualization = async ({
   setIsLoading(false);
 };
 
+const createOtelVisualizationMetaData = (
+  documentName: string,
+  visualizationType: string,
+  startTime: string,
+  endTime: string,
+  queryData: object
+) => {
+  return {
+    name: documentName,
+    description: '',
+    query: '',
+    type: visualizationType,
+    subType: PPL_METRIC_SUBTYPE,
+    metricType: OTEL_METRIC_SUBTYPE,
+    selected_date_range: {
+      start: startTime,
+      end: endTime,
+      text: '',
+    },
+    selected_timestamp: {
+      name: 'timestamp',
+      type: 'timestamp',
+    },
+    selected_fields: {
+      text: '',
+      tokens: [],
+    },
+    userConfigs: {
+      layout: dynamicLayoutFromQueryData(queryData),
+    },
+  };
+};
+
+export const fetchAggregatedBinCount = (
+  minimumBound: string,
+  maximumBound: string,
+  startTime: string,
+  endTime: string,
+  documentName: string,
+  http: CoreStart['http'],
+  selectedOtelIndex: string,
+  setIsError: React.Dispatch<React.SetStateAction<VizContainerError>>,
+  setIsLoading: React.Dispatch<React.SetStateAction<boolean>>
+) => async () => {
+  return http
+    .post(`${OBSERVABILITY_BASE}/metrics/otel/aggregatedBinCount`, {
+      body: JSON.stringify({
+        min: minimumBound,
+        max: maximumBound,
+        startTime,
+        endTime,
+        documentName,
+        index: selectedOtelIndex,
+      }),
+    })
+    .catch((error: Error) => {
+      const errorMessage = JSON.parse(error.body.message);
+      setIsError({
+        errorMessage: errorMessage.error.reason || 'Issue in fetching bucket count',
+        errorDetails: errorMessage.error.details,
+      });
+      console.error(error.body);
+    })
+    .finally(() => {
+      setIsLoading(false);
+    });
+};
+
+export const fetchSampleOTDocument = (
+  selectedOtelIndex: string,
+  http: CoreStart['http'],
+  documentName: string
+) => async () => {
+  return http
+    .post(`${OBSERVABILITY_BASE}/metrics/otel/sampleDocument`, {
+      body: JSON.stringify({
+        documentName,
+        index: selectedOtelIndex,
+      }),
+    })
+    .catch((error) => console.error(error));
+};
+
+export const extractIndexAndDocumentName = (metricString: string): [string, string] | null => {
+  const pattern = /\[Otel Metric\]\s(\S+?-\S+?)\.(\S+)/;
+  const match = metricString.match(pattern);
+
+  if (match) {
+    const index = match[1];
+    const documentName = match[2];
+    return [index, documentName];
+  } else {
+    return null;
+  }
+};
+
+export const renderOpenTelemetryVisualization = async ({
```

Comment:
Should we move this function and supporting functions to `metrics/utils.tsx`

Reply:
Kept this function here because similar functions like `renderCatalogVisualization` and `renderSavedVisualization` are in this file

Reply:
Can we reuse/update the `renderCatalogVisualization` function?

Reply:
We can't because both of them use different query language, metaData, and different visualizations as well. Not a feasible option as both are different